posts/squidpy-essentials/squidpy-essentials.qmd

```diff
@@ -0,0 +1,1071 @@
+│               ├── Run_86906312-688f-4f31-9fc5-48b04db5f958_<slide>_Beta15_ExptConfig.txt
+│               ├── SampleSlide_PlaneFit_<slide>_History.csv
+│               ├── Shading
+│               ├── c902.fovs.csv
+│               ├── latest.af.fovs.csv
+│               └── latest.fovs.csv
+├── flatFiles
+│   └── <flow cell>
+│       ├── <flow cell>_exprMat_file.csv.gz
+│       ├── <flow cell>_fov_positions_file.csv.gz
+│       ├── <flow cell>_metadata_file.csv.gz
+│       └── fov_positions_formatted.csv
+
+```
+
+So in order to read in image data into squidpy, we must first rearrange our folders. 
+We can do this manually by moving, copying, or linking files to match the expected
+format. Normally I am a big fan of symbolic links for just this situation. However, 
+I have noticed that `read.nanostring` can have an error when using symbolic links
+for some of the files (error not shown). So for simplicity I will make a copy of the 
+data and place them into a new folder `sample_dir_formatted`.
+
+```{default filename="Terminal"}
+#| eval: false
+
+# Note that this code works for a single slide. If exporting
+# multiple slides, some modification might be necessary.
+
+# cd to parent directory
+
+# create a directory adjacent to RawData and flatFiles
+mkdir -p sample_dir_formatted && cd $_
+
+# Add flat files
+for file in $(ls ../flatFiles/*/*csv*); do cp $file ./; done
+
+# Add folders
+cp -r ../RawFiles/*/*/CellStatsDir/CellComposite ./
+cp -r ../RawFiles/*/*/CellStatsDir/CellOverlay ./
+
+mkdir -p CellLabels
+for file in $(ls ../RawFiles/*/*/CellStatsDir/FOV*/CellLabels*); do cp $file ./CellLabels/ ; done
+rm ./CellLabels/._Cell*
+  
+mkdir -p CompartmentLabels
+for file in $(ls ../RawFiles/*/*/CellStatsDir/FOV*/CompartmentLabels*); do cp $file ./CompartmentLabels/ ; done
+```
+
+
+The file structure of `sample_dir_formatted` now looks like this:
+
+```{default filename="Terminal"}
+#| eval: false
+
+tree -L 2
+
+.
+├── <flow cell>_exprMat_file.csv.gz
+├── <flow cell>_fov_positions_file.csv.gz
+├── <flow cell>_metadata_file.csv.gz
+├── CellComposite
+│   ├── CellComposite_F001.jpg
+|   ├── ...
+│   └── CellComposite_F064.jpg
+├── CellLabels
+│   ├── CellLabels_F001.tif
+|   ├── ...
+│   └── CellLabels_F064.tif
+├── CellOverlay
+│   ├── CellOverlay_F001.jpg
+|   ├── ...
+│   └── CellOverlay_F064.jpg
+└── CompartmentLabels
+    ├── CompartmentLabels_F001.tif
+    ...
+    └── CompartmentLabels_F064.tif
+
+```
+
+
+:::{.callout-note}
+If your flow cell was created prior to AtoMx v1.3.2, the CellComposite
+jpg files may not show the composites but instead just blank images. 
+They will still load into squidpy but will not show the expected tissue structure.
+This issue was patched for samples created in v1.3.2+. If you have blank images and
+would like to replace them with composite images, check out [this post](../composite-images/making-composite-images.qmd){target="_blank"}. 
+:::
+
+
+## Read expression and image data into squidpy
```

Comment:
It's both _expression_ and _iamge_ analysis. I added a comment in the code to tell the reader which part is similar to the code they've seen before. It's not _identical_ since the name of the object is different. 096498969fe32f127cae3e1fa3fd28eb599d155b